src/main/java/com/mongodb/spark/sql/connector/schema/RowToBsonDocumentConverter.java

```diff
@@ -248,4 +304,20 @@ private boolean parseJsonData(final String data) {
         throw new AssertionError("Unexpected value: " + convertJson);
     }
   }
+
+  private static InternalRowToRowFunction createInternalRowToRowFunction(final StructType schema) {
+    try {
+      return new InternalRowToRowFunction(schema);
+    } catch (Exception e) {
+      throw new DataException(
+          format(
+              "Unable to create InternalRow to Row Function using %s. Errors: %s",
+              schema, e.getMessage()),
+          e);
+    }
+  }
+
+  interface ObjectToBsonElement extends Function<Object, BsonElement>, Serializable {}
+
+  interface ObjectToBsonValue extends Function<Object, BsonValue>, Serializable {}
```

Comment:
We don't seem to test directly whether the objects of types `ObjectToBsonElement`, `ObjectToBsonValue` that we create can actually be serialized. How sure are we that they are?

A quick glance suggests that the `ObjectToBsonElement` created in `dataTypeToBsonElement` may not be serializable in practice because it captures `field` of the `StructField` type, which does not seem to be `Serializable`. It also indirectly captures `ConvertJson`, which is also not `Serializable`.

Reply:
By using them in Spark (via the integration tests) they are tested - (Which lead me to create the interfaces for functions in the first place).

`StructField` is a Scala case class and is serializable.
`ConvertJson` is an enum so is inherently serializable.

Reply:
Terrible, I forgot that all enums are of the `Serializable` type.